test/EndToEndSpec.hs

```diff
@@ -0,0 +1,38 @@
+module EndToEndSpec (spec) where
+
+import           Test.Hspec                                    (Spec, it, shouldBe)
+import           Data.Text                                     (Text)
+import qualified Data.Aeson                                    as JSON
+import qualified Data.Aeson.Types                              as JSON.T
+import qualified Language.PureScript.CoreFn.FromJSON           as CoreFn
+import           Language.PureScript.CoreFn.Module             (Module)
+import           Language.PureScript.CoreFn.Ann                (Ann)
+import           Language.PureScript.Scheme.CodeGen.Transpiler (moduleToScheme)
+import           Language.PureScript.Scheme.CodeGen.Printer    (printScheme)
+import           Language.PureScript.Scheme.CodeGen.Optimizer  (runOptimizations)
+
+readJSONFile :: FilePath -> IO (Module Ann)
+readJSONFile path = do
```

Comment:
This is duplicated code - you should share it. Also the name is not that accurate, since it's more of a `readModuleFromJson`